### PR TITLE
chore: publish should retry at least once

### PR DESCRIPTION
**Pull request by markwhitfeld**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

During the last publish (v3.5) there was a glitch that caused one package to fail to be published.
We should retry at least once

## What is the new behavior?
Retry pcakage publishing at least once.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: tools/utils.ts [hunk 1]
```diff
@@ -47,14 +47,22 @@ export function execute(script: string, options: ExecOptions = {}): Promise<stri
...
+      await publishPackage(pack, version, tag);
     }
   }
 }
+async function publishPackage(pack: Package, version: any, tag: string) {
+  const packageDescription = `${pack.buildPath} ${version} @${tag}`;
+  try {
+    const script = `yarn publish --access public --non-interactive --no-git-tag-version --new-version ${version} --tag ${tag}`;
+    const output = await execute(script, { cwd: pack.buildPath });
+    console.log(`Published ${packageDescription} /r/n -> ${output}`);
```
Comment (splincode) [inline tools/utils.ts]: `/r/n`, maybe` \r\n` ?

Comment (markwhitfeld) [reply tools/utils.ts]: I just wrapped the code in a method and didn't change that.
It works, so I think we should leave it as is ;-)

**Comment by markwhitfeld** [comment]
Rebased and force pushed to get build to run again for this.

File: tools/utils.ts [hunk 2]
```diff
@@ -47,14 +47,22 @@ export function execute(script: string, options: ExecOptions = {}): Promise<stri
 export async function publishAllPackagesToNpm(version: any, tag: string) {
   const packages = getPackages();
   for (const pack of packages) {
-    const packageDescription = `${pack.buildPath} ${version} @${tag}`;
     try {
```
Comment (splincode) [inline tools/utils.ts]: What do you think about ?)

```ts
publishPackage(pack, version, tag)
  .catch(function oneRetry() { 
      publishPackage(pack, version, tag)
        .catch((err: Error) => console.error('Can't republish', err))
});
```

Comment (splincode) [reply tools/utils.ts]: forgive) that's incorrect for `for (const pack of packages) {`

**Pull request merged**